src/libs/metadata.c

```diff
@@ -107,6 +105,14 @@ static void _text_set_italic(GtkTextView *textview, const gboolean italic)
     gtk_text_buffer_remove_tag_by_name(buffer, "italic", &start, &end);
 }
 
+static void _set_text_buffer(GtkTextBuffer *buffer, const char *text)
+{
+  gulong handler = GPOINTER_TO_INT(g_object_get_data(G_OBJECT(buffer), "tv_handler"));
+  g_signal_handler_block(buffer, handler);
```

Comment:
Would something like `g_signal_handlers_block_matched(G_OBJECT(buffer), G_SIGNAL_MATCH_FUNC, 0, NULL, NULL, _textbuffer_changed, NULL);` not work here? (with matching `g_signal_handlers_unblock_matched` below)? No need to store and retrieve tv_handler. And somewhat easier to read (imvho). I haven't tried to compile, mind!

Reply:
Works as well. Done.